src/main/java/com/conveyal/datatools/common/utils/AWSUtils.java

```diff
@@ -115,4 +123,61 @@ public static String downloadFromS3(AmazonS3 s3, String bucket, String filename,
             return SparkUtils.formatJSON("url", url.toString());
         }
     }
+
+    /**
+     * Create credentials for a new session for the provided IAM role. The primary AWS account for the Data Tools
+     * application must be able to assume this role (e.g., through delegating access via an account IAM role
+     * https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html). The credentials can be
+     * then used for creating a temporary S3 or EC2 client.
+     */
+    public static AWSStaticCredentialsProvider getCredentialsForRole(String role, String sessionName) {
+        if (role == null) return null;
+        STSAssumeRoleSessionCredentialsProvider.Builder builder = new STSAssumeRoleSessionCredentialsProvider.Builder(role, "test");
+        STSAssumeRoleSessionCredentialsProvider sessionProvider = builder.build();
+        AWSSessionCredentials credentials = sessionProvider.getCredentials();
+        return new AWSStaticCredentialsProvider(new BasicSessionCredentials(
+            credentials.getAWSAccessKeyId(), credentials.getAWSSecretKey(),
+            credentials.getSessionToken()));
+    }
+
+    /**
+     * Shorthand method to obtain an EC2 client for the provided role ARN. If role is null, the default EC2 credentials
+     * will be used.
+     */
+    public static AmazonEC2 getEC2ClientForRole (String role) {
+        AWSStaticCredentialsProvider credentials = getCredentialsForRole(role, "ec2 client");
+        return AmazonEC2Client.builder().withCredentials(credentials).build();
```

Comment:
Perhaps to avoid repeating ourselves, this could call the `getEC2ClientForCredentials` method.